### PR TITLE
[WIP] x86 Conda recipe with DALI TF plugin

### PR DESCRIPTION
**Pull request by jantonguirao**
#### Why we need this PR?
*Pick one*
- It fixes a bug *bug description*
- It adds new feature needed because *why we need this feature*
- Refactoring to improve *what*

#### What happened in this PR?
 - Explain solution of the problem, new feature added.
 - What was changed, added, removed?
 - What is most important part that reviewers should focus on?
 - Was this PR tested? How?
 - Were docs and examples updated, if necessary?

**JIRA TASK**: [DALI-XXXX]

### CODE REVIEW

File: conda/recipe/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+# limitations under the License.
+
+{% set build_version = "master" %}
+
+  package:
+    name: dali
+    version: {{ build_version }}
+
+  source:
+    git_url: https://github.com/jantonguirao/DALI/
```
Comment (jantonguirao) [inline conda/recipe/meta.yaml]: TODO: revert to upstream master

**Comment by jayfurmanek** [comment]
Do we not want to continue with: https://github.com/NVIDIA/DALI/pull/1156
Why the 2nd almost duplicate PR?

**Comment by jayfurmanek** [comment]
I see the script for the plugin script added. That's good!. We'll give it and try and report back with comments.

**Comment by JanuszL** [comment]
@jayfurmanek - this PR is rather a testing ground. In the end, it would be good to have only one Conda recipe in https://github.com/NVIDIA/DALI/pull/1156.

File: conda/recipe/build.sh [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+make install
+
+# set RPATH of backend_impl.so and similar to $ORIGIN, $ORIGIN$UPDIRS, $ORIGIN$UPDIRS/.libs
+PKGNAME_PATH=$PWD/dali/python/nvidia/dali
+find $PKGNAME_PATH -type f -name "*.so*" -o -name "*.bin" | while read FILE; do
+    UPDIRS=$(dirname $(echo "$FILE" | sed "s|$PKGNAME_PATH||") | sed 's/[^\/][^\/]*/../g')
+    echo "Setting rpath of $FILE to '\$ORIGIN:\$ORIGIN$UPDIRS:\$ORIGIN$UPDIRS/.libs'"
+    patchelf --set-rpath "\$ORIGIN:\$ORIGIN$UPDIRS:\$ORIGIN$UPDIRS/.libs" $FILE
+    patchelf --print-rpath $FILE
+done
```
Comment (aalugore) [inline conda/recipe/build.sh]: Why is this RPATH section above required? (This question is more for my own enlightenment than anything)

Comment (aalugore) [inline conda/recipe/build.sh]: Why is this RPATH section above required? (Question is more for my own enlightenment than anything.)

Comment (jantonguirao) [reply conda/recipe/build.sh]: @aalugore This is required to be able to relocate the libraries once they are installed in the python package.
There are dependencies between the couple of libraries we provide and during link-time, the libraries are found in your build directory. Setting the rpath we are saying that this library should look for its dependencies in its current dir and above directories (e.g. "." and ".."). It is particularly necessary when the build is done in a different machine from the one you are installing the package.

**Pull request closed without merging**